### ISSUE TITLE
Adibag attempts to use ItemRack:RegisterExternalEventListener before Itemrack has initialized

### ISSUE THREAD

**Issue by MatthewBerkvens** [Rottenbeer/ItemRack#148]
Latest versions of both addons on 11/03 (will edit version numbers later)
https://i.imgur.com/wRXUtXr.png
https://i.imgur.com/9EwtkDN.png

I don't know who is responsible for fixing this, created an issue on both repos
https://github.com/AdiAddons/AdiBags/issues/558

**Comment by Rottenbeer** [comment]
@wardz could you have a look please?

**Comment by wardz** [comment]
Seems like the variable needs to be initialized earlier. You can just move https://github.com/Rottenbeer/ItemRack/blob/a73c04eda5ad1ad1c4c6df3c846d4374ff924fdc/ItemRack/ItemRack.lua#L147-L148 outside/above of the `ItemRack.InitEventHandlers` function.

**Comment by Rottenbeer** [comment]
@MatthewBerkvens test this please: https://github.com/Rottenbeer/ItemRack/tree/fix-adibags-race

**Comment by EnvoyMajeh** [comment]
I posted in the [AdiBags-ItemRackSets](https://github.com/apottere/AdiBags-ItemRackSets/issues/2) issues already, but I'll put this here in case it helps anyone:
Seems that ItemRack:RegisterExternalEventListener gets loaded later so ItemRack needs to finish loading before it can be used.
Not sure if this is the ideal fix, but I was able to get mine working again by changing a little bit in the Adibags-ItemRackSets.lua file -
On line 35/36 I changed:
```
ItemRack:RegisterExternalEventListener("ITEMRACK_SET_SAVED", itemRackUpdated)
ItemRack:RegisterExternalEventListener("ITEMRACK_SET_DELETED", itemRackUpdated)
```

to this:
```
local IREvents = CreateFrame("Frame", nil)
IREvents:RegisterEvent("ADDON_LOADED")
IREvents:SetScript("OnEvent", function(self, event, addon)
    if (addon == "ItemRack") then
        ItemRack:RegisterExternalEventListener("ITEMRACK_SET_SAVED", itemRackUpdated)
        ItemRack:RegisterExternalEventListener("ITEMRACK_SET_DELETED", itemRackUpdated)
    end
end)
```

**Comment by MatthewBerkvens** [comment]
> @wardz Thanks for having a look at this
> 
> @MatthewBerkvens test this please: https://github.com/Rottenbeer/ItemRack/tree/fix-adibags-race

Works, although the solution posted above seems preferable.

**Comment by Rottenbeer** [comment]
@MatthewBerkvens yeah maybe, but the solution above needs to be taken by the AdiBags dev.

**Comment by Rottenbeer** [comment]
I released 3.54 with the workaround. lmk if adibads pushed a fixed for this and we can revert the workaround.
